Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift

```diff
@@ -69,28 +69,33 @@ extension V1.Response {
         let authenticator: V1.Response.Authenticator
     }
 
-    func authenticatorState(for authenticatorId: String?) -> IDXClient.Authenticator.State {
-        guard let authenticatorId = authenticatorId else {
-            return .normal
-        }
-        
-        if currentAuthenticatorEnrollment?.value.id == authenticatorId {
-            return .enrolling
-        }
-        
-        else if currentAuthenticator?.value.id == authenticatorId {
-            return .authenticating
-        }
-        
-        else if authenticatorEnrollments?.value.first(where: { (authenticator) -> Bool in
-            authenticator.id == authenticatorId
-        }) != nil {
-            return .enrolled
-        }
+    func authenticatorState(for authenticators: [Authenticator],
+                            in jsonPaths: [String]) -> IDXClient.Authenticator.State
```

Comment:
This function is complex enough that it should probably be unit tested in various permutations?